modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundChannelFlowEvent.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c)2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.kernel.disruptor.inbound;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.kernel.AndesContext;
+import org.wso2.andes.kernel.disruptor.DisruptorEventCallback;
+
+import java.util.UUID;
+
+/**
+ * InboundChannelFlow event is used when a channel suspend/resume request is received. Once received, andes
+ * subscription manager is notified of the flow state to be updated.
+ */
+public class InboundChannelFlowEvent implements AndesInboundStateEvent {
+
+    private static Log log = LogFactory.getLog(InboundAndesChannelEvent.class);
+
+    /**
+     * Supported state events.
+     */
+    public enum EventType {
+
+        /**
+         * channel flow event signals the channel to be suspended/resumed.
+         */
+        CHANNEL_FLOW_EVENT
```

Comment:
Do we need an event type here? InboundChannelFlowEvent only has a single event type